Framework/PythonInterface/test/python/plugins/algorithms/WorkflowAlgorithms/ReflectometryILLSumForegroundTest.py

```diff
@@ -155,13 +155,66 @@ def testWavelengthRange(self):
         }
         alg = create_algorithm('ReflectometryILLSumForeground', **args)
         assertRaisesNothing(self, alg.execute)
-        ws = alg.getProperty('OutputWorkspace').value
-        self.assertEquals(ws.getNumberHistograms(), 1)
-        Xs = ws.readX(0)
+        out = alg.getProperty('OutputWorkspace').value
+        self.assertEquals(out.getNumberHistograms(), 1)
+        Xs = out.readX(0)
         self.assertGreater(len(Xs), 1)
         self.assertGreater(Xs[0], xMin)
         self.assertLess(Xs[-1], xMax)
 
+    def testWavelengthRangeDefault(self):
+        ws = illhelpers.create_poor_mans_d17_workspace()
+        illhelpers.refl_rotate_detector(ws, 1.2)
+        beamPosWS = illhelpers.refl_create_beam_position_ws('beamPosWS', ws, 1.2, 128)
+        ws = illhelpers.refl_preprocess('ws', ws, beamPosWS)
+        args = {
+            'InputWorkspace': ws,
+            'OutputWorkspace': 'foreground',
+            'rethrow': True,
+            'child': True
+        }
+        alg = create_algorithm('ReflectometryILLSumForeground', **args)
+        assertRaisesNothing(self, alg.execute)
+        out = alg.getProperty('OutputWorkspace').value
+        self.assertEquals(out.getNumberHistograms(), 1)
+        Xs = out.readX(0)
+        self.assertGreater(len(Xs), 1)
+        self.assertGreater(Xs[0], 0.)
+        self.assertLess(Xs[-1], 30.)
+
+    def testNoDirectForegroundAndSumInQRaises(self):
+        ws = illhelpers.create_poor_mans_d17_workspace()
+        illhelpers.refl_rotate_detector(ws, 1.2)
+        beamPosWS = illhelpers.refl_create_beam_position_ws('beamPosWS', ws, 1.2, 128)
+        ws = illhelpers.refl_preprocess('ws', ws, beamPosWS)
+        args = {
+            'InputWorkspace': ws,
+            'OutputWorkspace': 'foreground',
+            'SummationType': 'SumInQ',
+            'rethrow': True,
+            'child': True
+        }
+        alg = create_algorithm('ReflectometryILLSumForeground', **args)
+        with self.assertRaises(AssertionError) as contextManager:
```

Comment:
This could be simplified and made more robust by using `assertRaisesRegexp`:
```python
self.assertRaisesRegexp(RuntimeError, 'Direct foreground workspace is needed for summing in Q.', alg.execute)
```
Please check that `execute` actually raises a `RuntimeError`.